pkg/network/setup/hotplugcontroller.go

```diff
@@ -20,9 +20,60 @@
 package network
 
 import (
+	"fmt"
+	"strings"
+
 	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
 )
 
+type ConcreteController struct {
+	ifaceCacheFactory cache.CacheCreator
+	vmi               *v1.VirtualMachineInstance
+	nsFactory         nsFactory
+}
+
+func NewInterfaceController(cacheFactory cache.CacheCreator, ns nsFactory) *ConcreteController {
+	return &ConcreteController{
+		ifaceCacheFactory: cacheFactory,
+		nsFactory:         ns,
+	}
+}
+
+func (c *ConcreteController) HotplugIface(iface *v1.Network, launcherPID int) error {
+	log.Log.V(4).Infof("creating networking infra for network: %s", iface.Name)
+	if err := c.nsFactory(launcherPID).Do(func() error {
+		return NewVMNetworkConfigurator(
+			c.vmi,
+			c.ifaceCacheFactory,
+		).CreatePodAuxiliaryInfra(launcherPID, iface.Name)
+	}); err != nil {
+		return fmt.Errorf("setup failed, err: %w", err)
+	}
+
+	return nil
+}
+
+func (c *ConcreteController) HotplugIfaces(vmi *v1.VirtualMachineInstance, launcherPID int) error {
+	c.vmi = vmi
+
+	networksToHotplug := NetworksToHotplugWhosePodIfacesAreReady(vmi)
```

Comment:
Because of the predicate I am passing to this func; it ensure the interface status does not feature the `domain` infosource. You're asking about this in https://github.com/kubevirt/kubevirt/pull/6852#discussion_r1073741285